Formula/amp.rb

```diff
@@ -38,22 +43,30 @@ def install
   end
 
   test do
-    input, _, wait_thr = Open3.popen2 "script -q /dev/null"
-    input.puts "stty rows 80 cols 43 && #{bin}/amp test.txt"
-    sleep 1
-    # switch to insert mode and add data
-    input.putc "i"
-    sleep 1
-    input.puts "test data"
-    # escape to normal mode, save the file, and quit
-    input.putc "\e"
-    sleep 1
-    input.putc "s"
-    sleep 1
-    input.putc "Q"
+    require "pty"
+    require "io/console"
+
+    PTY.spawn(bin/"amp", "test.txt") do |r, w, _pid|
+      r.winsize = [80, 43]
+      sleep 1
+      # switch to insert mode and add data
+      w.write "i"
+      sleep 1
+      w.write "test data"
+      sleep 1
+      # escape to normal mode, save the file, and quit
+      w.write "\e"
+      sleep 1
+      w.write "s"
+      sleep 1
+      w.write "Q"
+      begin
+        r.read
+      rescue Errno::EIO
+        # GNU/Linux raises EIO when read is done on closed pty
+      end
```

Comment:
Not exactly sure why `r.read` is necessary (maybe it stalls until process finishes or deals with buffering?), but without this the test file is often not written for `assert_match` check.